File: progChallenges/september/bitsets.py
#!/usr/bin/env python3

elements = (1,2,4)
#initialize bitst
bitset = 0

# add elements to bitset
for i in elements:
    bitset = bitset | (1 << i)

formatted = f'{bitset:08b}'
print(f'{bitset:x}')        # hex
print(f'{bitset:08b}')      # binary 

# test for elements  bitset
for i in range(6):
    if bitset & 1<<i:
        print(f'{i} in bitset')

# remove 
for i in elements:
    bitset = bitset & ~(1<<i)

print(f'{bitset:08b}')      # binary 

File: progChallenges/september/bitsets.c
#include <stdint.h>
#include <stdio.h>

int main(int argc, char* argv[]) {
    int elements[] = {1,2,4,0};
    //initialize bitset
    uint8_t bitset = 0;

    // add elements to bitset, use or to add
    // elements[i] has 0 as a sentinel, that's why we use elements[i] as the condition
    for (size_t i = 0; elements[i]; i++) {
        // 7 6 5 4 3 2 1 0
        // 0 0 0 0 0 0 0 0 [0]
        // 0 0 0 0 0 0 1 0 [1]
        bitset = bitset |  1 << elements[i];
    }
    // 0 0 0 1 0 1 1 0
    // %d decimal
    // %x hex
    // %c char 
    printf("%d\n",bitset);

    // test for elements in bitset , use and to check
    for (size_t i = 0; i < 6; i++) {
        if (bitset & (1<<i)){
            printf("%lu in bitset\n",i);
        }
    }

    //remove elements from bitset, USE AND WITH COMPLEMENT
        //could use XOR (don't use the xor twice, otherwise it will add it back)
    for (size_t i = 0; elements[i]; i++) {
        bitset = bitset &  ~(1 << elements[i]);
    }
    printf("%d\n",bitset);

}